### ISSUE TITLE
Sphinx 2.0 breaks some unit tests

### ISSUE THREAD

**Issue by bnmnetp** [RunestoneInteractive/RunestoneServer#1221]
Sphinx 2.0 came out, and how warnings are handled has changed.  I updated the runestone build code to follow their advice on using the sphinx.util.logging but there still seems to be a problem.  Especially in the test that tests preview -- and expects a warning.

Retreating to sphinx 1.8.5 fixes the problem.  This does not seem to affect the Travis build (maybe that is 2.7) but it definitely kills my dev environment.

@bjones1 I will be happy to try to dig into this as a way of getting more familiar with pytest, but may need some advice.

**Comment by bjones1** [comment]
Sphinx 2.0 did make some significant changes. I'll be happy to talk a look. I'm on the road until Tuesday, but will hopefully be able to play with it them.

Sphinx 2.0 is Python 3 only, so it won't run on Travis, which tests under Python 2.7.

**Comment by bjones1** [comment]
Per our Slack discussion, @bnmnetp will fix this, probably through 8cd3c6f79868e6e676d84ef8d97f77e28d75b022.

**Comment by bnmnetp** [comment]
Well.... there were multiple factors...

1.  I simplified some parts of __main__.py -- which in my testing did not hurt anything, but in the way that the unittests run the runestone command broke pretty much everything.

2.  Currently there are still some problems with Sphinx 2.0.1 breaking a few of our tests.... I'm not sure why.

3.  In the meantime I've updated requirements.txt to use Sphinx 1.8.4 so that all the tests pass.

Now, CI is happy, but devs should still not be happy so that we can figure out why the remaining handful of tests are not working on Sphinx 2.X
